rest-api/test/unit_test/api_test/biobank_order_api_test.py

```diff
@@ -17,9 +19,315 @@ def setUp(self):
     self.participant_dao = ParticipantDao()
     self.participant_dao.insert(self.participant)
     self.summary_dao = ParticipantSummaryDao()
+    self.bio_dao = BiobankOrderDao()
     self.path = (
         'Participant/%s/BiobankOrder' % to_client_participant_id(self.participant.participantId))
 
+  def test_cancel_order(self):
+    self.summary_dao.insert(self.participant_summary(self.participant))
+    order_json = load_biobank_order_json(self.participant.participantId,
+                                         filename='biobank_order_2.json')
+    result = self.send_post(self.path, order_json)
+    full_order_json = load_biobank_order_json(self.participant.participantId,
+                                              filename='biobank_order_1.json')
+    _strip_fields(result)
+    _strip_fields(full_order_json)
+    self.assertEquals(full_order_json, result)
+
+    biobank_order_id = result['identifier'][1]['value']
+    path = self.path + '/' + biobank_order_id
+    request_data = {
+      "amendedReason": "Its all wrong",
+      "cancelledInfo": {
+        "author": {
+          "system": "https://www.pmi-ops.org/healthpro-username",
+          "value": "fred@pmi-ops.org"
+        },
+        "site": {
+          "system": "https://www.pmi-ops.org/site-id",
+          "value": "hpo-site-monroeville"
+        }
+      },
+      "status": "cancelled"
+    }
+    cancelled_order = self.send_patch(path, request_data=request_data,
+                                      headers={'If-Match': 'W/"1"'})
+    get_cancelled_order = self.send_get(path)
+    get_summary = self.summary_dao.get(self.participant.participantId)
+
+    self.assertEqual(get_summary.biospecimenSourceSiteId, None)
```

Comment:
You could assert that participant summary sample statuses and timestamps are unset here, too. (I know you've got DAO tests for that, too, but API tests are nice.)